charts/nr-k8s-otel-collector/values.yaml

```diff
@@ -0,0 +1,100 @@
+  # metrics by default. You can enable the target labels/annotations metrics to be monitored by using the metricLabelsAllowlist/metricAnnotationsAllowList options described [here](https://github.com/prometheus-community/helm-charts/blob/159cd8e4fb89b8b107dcc100287504bb91bf30e0/charts/kube-state-metrics/values.yaml#L274) in
+  # your Kubernetes clusters.
+  enabled: true
+
+replicaCount: 1
+
+image:
+  repository: otel/opentelemetry-collector-contrib
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "0.91.0"
+
+# -- This is the New Relic endpoint to send data to.
+newRelicEndpoint: "https://otlp.nr-data.net"
+# -- Name of the Kubernetes cluster monitored. Mandatory. Can be configured also with `global.cluster`
+cluster: ""
+# -- This set this license key to use. Can be configured also with `global.licenseKey`
+licenseKey: ""
+# -- In case you don't want to have the license key in you values, this allows you to point to a user created secret to get the key from there. Can be configured also with `global.customSecretName`
+customSecretName: ""
+# -- In case you don't want to have the license key in you values, this allows you to point to which secret key is the license key located. Can be configured also with `global.customSecretLicenseKey`
+customSecretLicenseKey: ""
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+podAnnotations: {}
+
+podSecurityContext: {}
+  # fsGroup: 2000
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  type: ClusterIP
+  port: 80
+
+ingress:
+  enabled: false
+  className: ""
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+resources: {}
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
```

Comment:
my advice is avoiding having the very same resources for the daemonSet and for the deployment. Fixing it in the feature could be complex

Reply:
Yeah, we will have to do a load test to confirm these values after we cut down on the amount of metrics being sent over, but can you clarify why fixing the feature could be complex?

Reply:
Right now we have a deployment A and adeamonSet B

They share the same `resources `, however in a cluster with 200 nodes the gateway could end up consuming 4-5x than a single instance on a node since it receive and process the traffic coming from all over the cluster (and scrapes KSM)
For this reason, I think that is safer to have most of the option under the specific component:
```
depoyment :   <-- not sure about the name 😄 
  enabled: {}
  podannotations: {}
  extraEvnVars: {}
  resources: {}
  [...]
daemonSet:
  enabled: {}
  podannotations: {}
  extraEvnVars: {}
  resources: {}
  [...]
```
Similar to what we have already [here](https://github.com/newrelic/nri-kubernetes/blob/main/charts/newrelic-infrastructure/values.yaml)

Otherwise you could endup giving a lot of resources to all instances, just because the single one from the deployment needs more

Reply:
hmm, that makes a lot of sense, thanks. I'll do this in a followup PR as we work through load and ingest testing.